### ISSUE TITLE
New Rule: Use multiline comments instead of mulitple lines of single line comments

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#2394]
From GitLab SQL style guide: https://about.gitlab.com/handbook/business-technology/data-team/platform/sql-style-guide/#commenting

```
- When making single line comments in a model use the -- syntax
- When making multi-line comments in a model use the /* */ syntax
```

This is nice because you can easily write and expand multiline comments without having to add the `--` to the start of each line.

This rule would detect if you have multiple consecutive lines of inline comments (on their own line) and replace it with a block comment.

Bad:
```sql
-- This is some big long
-- description of some code
-- that I am explaining...
SELECT foo FROM bar;
```

Good:
```sql
/*
This is some big long
description of some code
that I am explaining...
*/
SELECT foo FROM bar;
```

The Brooklyn Data co. SQL style guide is a bit more harsh in that it always requires the use of mutiline comment blocks so maybe that can be an config option (def not enabled by default): https://github.com/brooklyn-data/co/blob/main/sql_style_guide.md#always-use---comment-syntax

**Comment by barrywhart** [comment]
I have always preferred multiple single-line comments, because it avoids the likelihood that someday, I will forget to close the comment. 🙀

**Comment by jpy-git** [comment]
@barrywhart Could be another config option to always use single-line comments?

**Comment by barrywhart** [comment]
Sounds good to me! It's funny how much people's opinions differ.

**Comment by jpy-git** [comment]
Yeah I went through all the main SQL style guides and picked out the rules that I thought were the most widely applicable that we could use here.
So any further refinements/edge cases/additional options is appreciated so we can flesh them out before working on them 👍
